network/transport/v2/handlers.go

```diff
@@ -139,6 +142,31 @@ func (p *protocol) handleTransactionPayload(msg *TransactionPayload) error {
 	return p.payloadScheduler.Finished(ref)
 }
 
+func (p protocol) handleTransactionRangeQuery(peer transport.Peer, envelope *Envelope_TransactionRangeQuery) error {
+	msg := envelope.TransactionRangeQuery
+	cid := conversationID(msg.ConversationID)
+	if err := p.cMan.check(envelope); err != nil {
+		return err
+	}
+
+	if envelope.TransactionRangeQuery.Start >= envelope.TransactionRangeQuery.End {
+		return errors.New("invalid range query")
+	}
+
+	ctx := context.Background()
+	txs, err := p.state.FindBetweenLC(ctx, envelope.TransactionRangeQuery.Start, envelope.TransactionRangeQuery.End)
+	if err != nil {
+		return err
+	}
+
+	transactionList, err := p.collectTransactionList(ctx, txs)
+	if err != nil {
```

Comment:
The handleTransactionListQuery logs a warning here.

Reply:
Removed the log message there: I left it there because then the peer ID would get logged. But I think we need to log handler errors on a central spot including all metadata (peer ID and such): https://github.com/nuts-foundation/nuts-node/issues/973